File: frontend/services/teacherHourStats.ts
// [教质模块] 教员课时统计服务
import { api } from './api'

export interface TeacherHourSchedule {
  [key: string]: any
}

export interface TeacherHourStatsPayload {
  campus: string
  year: number
  month: number
  schedule: TeacherHourSchedule
  teacher: string
}

export const fetchTeacherHourStats = async (params: {
  campus: string
  year: number
  month: number
  teacher: string
}) => {
  const res = await api.get<{
    campus: string
    year: number
    month: number
    schedule: any
    teacher: string | null
  }>('/teacher-hour-stats', { params })
  return res.data
}

export const saveTeacherHourStats = async (payload: TeacherHourStatsPayload) => {
  const res = await api.post('/teacher-hour-stats', payload)
  return res.data
}

File: backend/app/schemas/teacher_hour_stats.py
"""
Pydantic schemas for teacher hour monthly stats
"""

from typing import Annotated, Any, Dict, Optional

from pydantic import BaseModel, ConfigDict, Field


class TeacherHourStatsBase(BaseModel):
    campus: str
    year: Annotated[int, Field(ge=2000, le=2100)]
    month: Annotated[int, Field(ge=1, le=12)]


class TeacherHourStatsPayload(TeacherHourStatsBase):
    schedule: Dict[str, Any] = Field(default_factory=dict)
    teacher: str


class TeacherHourStatsResponse(TeacherHourStatsBase):
    schedule: Optional[Dict[str, Any]] = None
    teacher: Optional[str] = None

    model_config = ConfigDict(
        from_attributes=True,
    )
